### ISSUE TITLE
GraphQLContext annotation adds the query to the top-level query as well.

### ISSUE THREAD

**Issue by nukesz** [leangen/graphql-spqr#15]
As in the [SocialNetworkQuery sample](https://github.com/leangen/graphql-spqr-samples/blob/master/src/main/java/io/leangen/spqr/samples/demo/query/annotated/SocialNetworkQuery.java) I can attach the query to an existing domain without changing the class itself. It works, but it also adds the query to the top-level "QUERY_ROOT" also by default (as it sometimes make sense to pass the context explicitly). 
It would be nice to have a way to control this behavior and disable if it's possible.

**Comment by depauwjimmy** [comment]
I can say that it would be nice to control this behavior. A nice-to-have indeed.

**Comment by tleveque23** [comment]
I have the same request. In my test code I have users with orders. My orders are listed using the user as argument. I don't want to expose this method in my root.

**Comment by p-maks** [comment]
+ 1 for this

**Comment by chrisflatley** [comment]
Feels like the default behaviour should be to not add something to the QUERY_ROOT. 

Or perhaps have a GraphQLContext which adds to the root and at GraphQLExtension annotation which doesn't?

**Comment by kaqqao** [comment]
I've finally fixed this in master, will be released in v0.9.4, likely tomorrow.
The nested operations will not be added to QUERY_ROOT any longer. For now, there's no automatic way to get the old behavior, so creating an additional method without `@GraphQLContext` is needed.

I was trying to add the ability to get the old behavior if the same argument has both `@GraphQLContext` and `@GraphQLArgument`, but it ended up causing too much complexity for relatively little gain so I decided against it. If you see this feature as useful/important, drop a line.
